docker/indexer/stages/processing/processing.go

```diff
@@ -51,48 +50,32 @@ type FileResult struct {
 type Stage struct {
 	Storer  Storer
 	RepoHdl *storage.BucketHandle
+	Input   *pubsub.Subscription
 }
 
 // Run runs the stages and hashes all files for each incoming request.
-func (s *Stage) Run(ctx context.Context, input chan *preparation.Result) error {
-	wErr := make(chan error, workers)
-	sem := semaphore.NewWeighted(workers)
-	for {
-		if err := sem.Acquire(ctx, 1); err != nil {
-			return fmt.Errorf("failed to acquire semaphore: %v", err)
+func (s *Stage) Run(ctx context.Context) error {
+	s.Input.ReceiveSettings.MaxOutstandingMessages = 5
```

Comment:
nit: put 5 in a constant.

Reply:
Also, one thing to note is that in the past I found this was not respected unless `Synchronous` is set to `true`. What ended up happening is one worker would queue up most of the tasks while the other remaining available workers got nothing. 

However, according to the latest docs at https://pkg.go.dev/cloud.google.com/go/pubsub#ReceiveSettings, this seems to be fixed. Still worth double checking though to make sure it works as attended.

Reply:
I tried this on a GCE instance and the number of go routines reduced drastically to not setting it. Also with two receiving clients both got work to do.